torch/distributed/_sharded_tensor/api.py

```diff
@@ -1018,18 +985,20 @@ def reshard(self, resharding_spec: ShardingSpec) -> ShardedTensor:
         if not isinstance(resharding_spec, ChunkShardingSpec):
             raise NotImplementedError("Only ChunkShardingSpec supported for reshard.")
         sharding_dim = int(resharding_spec.dim)  # type: ignore[attr-defined]
-        world_size = dist.get_world_size(self.pg)
-        local_shards = torch.tensor_split(self.local_partial_shard, world_size)
+        if self.local_shard.size(sharding_dim) % self.process_group.size() != 0:
+            raise ValueError('World size need to divide the length of the dimension.')
```

Comment:
Why do we need this restriction?

Reply:
This is to answer @mrshenli 's previous comment. I guess this is one assumption we have for PartialTensor right? Otherwise, we will get uneven shape after `torch.chunk`?

Reply:
Discussed offline, we are going to add padding for these cases and I included in the filed GH issue: https://github.com/pytorch/pytorch/issues/72022